### PR TITLE
Registry schema docs

### PR DESCRIPTION
**Pull request by GeekArthur**
This PR adds registry schema docs.

Signed-off-by: jingfu wang <jingfwan@redhat.com>

### CODE REVIEW

**Comment by GeekArthur** [comment]
@johnmcollier Please review this docs as well, thanks.

File: registry-schema.adoc [hunk 1]
```diff
@@ -0,0 +1,418 @@
+= Registry Schemas
+
+== extraDevfileEntries.yaml
+A file called extraDevfileEntries.yaml can be added under the root of the devfile registry source repository to add samples and stacks from other repositories to the registry. This file contains the location information on where the extra samples and stacks can be found.
+
+=== extraDevfileEntries.yaml schema
+- samples: schema[] - The sample list
+- stacks: schema[] - The stack list
+- name: string - The stack name
```
Comment (johnmcollier) [inline registry-schema.adoc:10]: Wondering if we need to make it clear that the items for `name` and below are for the `schema` type and not top level attributes. What do you think?

File: registry-schema.adoc [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+    tags: ["Python"]
+    projectType: python
+    language: python
+    git:
+      remotes:
+        origin: https://github.com/elsony/devfile-sample-python-basic.git
+----
+
+== index.json
+A file called index.json contains the metadata of stacks and samples in the registry, user can get the stack or sample generic infomration such as name, language and resources.
```
Comment (johnmcollier) [inline registry-schema.adoc]: We should maybe also state that it's an array of devfile stack index entries? Not sure how to phrase that. (Basically an array of the type that you defined down below)

File: registry-schema.adoc [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+    language: python
+    git:
+      remotes:
+        origin: https://github.com/elsony/devfile-sample-python-basic.git
+----
+
+== index.json
+A file called index.json contains the metadata of stacks and samples in the registry, user can get the stack or sample generic infomration such as name, language and resources. 
+
+=== index.json schema
```
Comment (johnmcollier) [inline registry-schema.adoc]: Maybe Devfile Index Schema? What do you think? I'm not sure, to be honest

**Review by johnmcollier — CHANGES_REQUESTED** [review]
Generally looks good. Just a couple nit picks / questions I had.

File: registry-schema.adoc (hunk 1 above)
Comment (GeekArthur) [reply registry-schema.adoc:10]: Let me try to modify it to make it more clear.

File: registry-schema.adoc (hunk 3 above)
Comment (GeekArthur) [reply registry-schema.adoc]: I update it to `index.json (devfile index) schema`, the reason I want to keep the index.json is:
1. To let user know the schema is used for index.json file
2. To make it consistent with `extraDevfileEntries.yaml schema`

File: registry-schema.adoc (hunk 2 above)
Comment (GeekArthur) [reply registry-schema.adoc]: I can emphasize that from data structure perspective.

**Comment by GeekArthur** [comment]
@johnmcollier Comments addressed, please review again, thanks.

**Review by johnmcollier — APPROVED** [review]

**Review by Nosense363 — COMMENTED** [review]
Learning to use.

**Pull request merged**